library/src/com/orm/util/ReflectionUtil.java

```diff
@@ -175,11 +175,30 @@ public static void setFieldValueFromCursor(Cursor cursor, Field field, Object ob
             Log.e("field set error", e.getMessage());
         }
     }
+    
+    /**
+     * Lookup a field in the specified class and all the superclasses
+     * 
+     * @param cls
+     * @param name
+     * @return
+     * @throws NoSuchFieldException
+     */
+    public static Field getField(Class<?> cls, String name) throws NoSuchFieldException {
+    	try {
+    		Field field = cls.getDeclaredField(name);
+    		return field;
+        } catch (NoSuchFieldException e) {
+        	if (cls.getSuperclass() != null)
+        		return getField(cls.getSuperclass(), name);
+        	throw e;
+        }
+    }
 
     public static void setFieldValueForId(Object object, Long value) {
 
         try {
-            Field field = object.getClass().getField("id");
+            Field field = getField(object.getClass(), "id");
```

Comment:
My guess is that this should've gone in another pull request.

As the code is currently implemented, the id is only set if the record object has defined an `id` instance variable. Otherwise, even if the record extends from SugarRecord, the method getId() returns null. 

I don't know if that was your intention but I feel that is a big change from the behavior up to version 1.3